docs/coordinates.rst

```diff
@@ -0,0 +1,444 @@
+Coordinates can be added to this object if and when the user sees fit.
+Let's attach a scalar global coordinate to ``my_cube`` representing some kind of distance.
+We do this by supplying the coordinate's name, physical type and value via the `~ndcube.GlobalCoords.add` method.
+
+.. code-block:: python
+
+  >>> import astropy.units as u
+  >>> my_cube.global_coords.add('distance', 'pos.distance', 1 * u.m)
+
+Because `~ndcube.GlobalCoords` allows multiple coordinates of the same physical type, a unique coordinate name must be provided.
+Furthermore the physical type must be a valid `IVOA UCD1+ controlled words <http://www.ivoa.net/documents/REC/UCD/UCDlist-20070402.html>`_ word.
+If one does not exist for your coordinate, prepend the type with ``custom:``.
+
+The value of the coordinate can be accessed by indexing the `~ndcube.GlobalCoords` instance with the coordinate name.
+
+.. code-block:: python
+
+  >>> my_cube.global_coords['distance']
+  <Quantity 1. m>
+
+The coordinate's physical type can be accessed via the `~ndcube.GlobalCoords.physical_types` `dict` property.
+
+.. code-block:: python
+
+  >>> my_cube.global_coords.physical_types['distance']
+  'pos.distance'
+
+Because `~ndcube.GlobalCoords` inherits from `~collections.abc.Mapping`, it contains a number of mixin methods similar to those of `dict`.
+
+.. code-block:: python
+
+  >>> list(my_cube.global_coords.keys())  # Returns a list of global coordinate names
+  ['distance']
+  >>> list(my_cube.global_coords.values())  # Returns a list of coordinate values
+  [<Quantity 1. m>]
+  >>> list(my_cube.global_coords.items())  # Returns a list of (name, value) pairs
+  [('distance', <Quantity 1. m>)]
+
+A common use case for `~ndcube.GlobalCoords` is associated with slicing (:ref:`cube_slicing`).
+In addition to tracking and updating the `~ndcube.NDCube.wcs` and `~ndcube.NDCube.extra_coords` objects, `~ndcube.NDCube`'s slicing infrastucture also identifies when the array axes to which a coordinate corresponds are dropped.
+The values of dropped coordinates at the position where the `~ndcube.NDCube` was sliced are stored in the `astropy.wcs.WCS` instance from where `~ndcube.GlobalCoords` can access and return them.
+
+.. code-block:: python
+
+  >>> my_2d_cube = my_cube[:, :, 0]
+  >>> my_2d_cube.array_axis_physical_types  # Note the wavelength axis is now gone.
+  [('custom:pos.helioprojective.lat', 'custom:pos.helioprojective.lon'),
+   ('custom:pos.helioprojective.lat', 'custom:pos.helioprojective.lon')]
+
+  >>> # The wavelength value at the slicing location is now in the GLobalCoords object.
+  >>> list(my_2d_cube.global_coords.keys())  # doctest: +SKIP
+  ['distance', 'em.wl']  # doctest: +SKIP
+  >>> my_2d_cube.global_coords.physical_types['em.wl']  # doctest: +SKIP
+  'em.wl'  # doctest: +SKIP
+  >>> my_2d_cube.global_coords['em.wl']  # doctest: +SKIP
+  <SpectralCoord 1e-9 m>
+
+.. _cube_coordinates:
+
+NDCube Coordinates
```

Comment:
Should this not be the first section in this document? It feels weird going from a WCS description to EC and GC. Would this then EC then GC be a better flow?

Reply:
Possibly.  Will give it some though before merging.